src/flash/com/longtailvideo/jwplayer/media/VideoMediaProvider.as

```diff
@@ -97,9 +99,10 @@ public class VideoMediaProvider extends MediaProvider {
     /** Load new media file; only requested once per item. **/
     override public function load(itm:PlaylistItem):void {
         setState(PlayerState.LOADING);
-        if (_item !== itm) {
+        if (_item !== itm || _complete) {
             setupVideo(itm);
             loadQuality();
+            _complete = false;
```

Comment:
Set this to false in `loadStream()`, since there are a couple of other ways that `_complete` can be reset (any call to `_stream.play` reloads the video).